### PR TITLE
x86,armsr: interpolate @GRUB_SERIAL@ into /etc/inittab

### PR DESCRIPTION
**Pull request by pprindeville**
On platforms where `/dev/ttyS0` isn't the console port, allow substituting in `$(CONFIG_GRUB_SERIAL)` as `@GRUB_SERIAL@` into `/etc/inittab`.

### CODE REVIEW

**Comment by pprindeville** [comment]
cc: @KanjiMonster

**Comment by pprindeville** [comment]
Since you all are interested in `armsr` support.

cc: @anta5010 @oskarirauta @1715173329

**Comment by anta5010** [comment]
cc: @mcbridematt

File: package/base-files/Makefile [hunk 1]
```diff
@@ -199,7 +199,8 @@ define Package/base-files/install
 		$(1)/usr/lib \
 		$(1)/usr/bin \
 		$(1)/sys \
-		$(1)/www \
+		$(1)/www
+	mkdir -p -m 750 \
```
Comment (jow-) [inline package/base-files/Makefile:203]: This change looks unrelated, please drop it from this PR.

Comment (pprindeville) [reply package/base-files/Makefile:203]: Moved to PR #13892.

File: target/linux/x86/base-files.mk [hunk 2]
```diff
@@ -0,0 +1,10 @@
+GRUB_SERIAL:=$(call qstrip,$(CONFIG_GRUB_SERIAL))
+ifeq ($(GRUB_SERIAL),)
+$(error This platform requires CONFIG_GRUB_SERIAL be set!)
+endif
+
+ifneq ($(GRUB_SERIAL),)
```
Comment (hauke) [inline target/linux/x86/base-files.mk:6]: Why do you check if `GRUB_SERIAL` is set here? It should error our before already.

Comment (pprindeville) [reply target/linux/x86/base-files.mk]: Dropped

File: target/linux/armsr/base-files/etc/inittab [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ::sysinit:/etc/init.d/rcS S boot
 ::shutdown:/etc/init.d/rcS K shutdown
 ttyAMA0::askfirst:/usr/libexec/login.sh
-ttyS0::askfirst:/usr/libexec/login.sh
+@GRUB_SERIAL@::askfirst:/usr/libexec/login.sh
```
Comment (KanjiMonster) [inline target/linux/armsr/base-files/etc/inittab:4]: https://github.com/openwrt/openwrt/blob/main/target/linux/armsr/image/Makefile#L20 and https://github.com/openwrt/openwrt/blob/main/target/linux/x86/image/Makefile#L21 currently assume `GRUB_SERIAL` can be empty, which implies this is a valid configuration.

Maybe instead of erroring out append/replace the whole line instead when `GRUB_SERIAL` is empty?

Comment (hauke) [reply target/linux/armsr/base-files/etc/inittab:4]: I think it is always set on the x86 and armsr target. I think we should unify this handling.

Comment (pprindeville) [reply target/linux/armsr/base-files/etc/inittab:4]: Resolved by making the above two files unconditional.
